data/doxygen.h

```diff
@@ -7,45 +7,83 @@
  *
  * \code {.cpp}
  * #include <taskomat/taskomat.h>
  *
  * using namespace task;
  *
- * // Create some steps
- * Step step1{ Step::type_action };
- * step1.set_script("a = 42");
+ * Sequence create_sequence()
+ * {
+ *     // Create some steps
+ *     Step step1{ Step::type_action };
+ *     step1.set_script("a = 42");
+ *
+ *     Step step2{ Step::type_action };
+ *     step2.set_script("print('Mary had a little lamb.')");
+ *
+ *     // Assemble the steps into a sequence
+ *     Sequence sequence{ "An example sequence" };
+ *     sequence.push_back(step1);
+ *     sequence.push_back(step2);
  *
- * Step step2{ Step::type_action };
- * step2.set_script("b = 13");
+ *     return sequence;
+ * }
  *
- * // Assemble the steps into a sequence
- * Sequence sequence;
- * sequence.push_back(step1);
- * sequence.push_back(step2);
+ * Sequence sequence = create_sequence();
+ * Context context; // A context may contain additional information for the sequence
  *
  * // Run the sequence
- * sequence.execute();
+ * sequence.execute(context);
+ * \endcode
+ *
+ * \section Running a sequence in a parallel thread
+ *
+ * Taskolib can start a sequence in a parallel thread via an \ref task::Executor Executor
+ * object. The main thread should frequently call \ref task::Executor::update update() on
+ * this object to get information about the status of the execution. No other
+ * synchronization is necessary.
+ *
+ * \code {.cpp}
+ * #include <taskomat/taskomat.h>
+ *
+ * using namespace task;
+ *
+ * Executor ex;
+ * Sequence sequence = create_sequence(); // get a Sequence from somewhere
+ * Context context;
+ *
+ * // Start executing a copy of the sequence in a separate thread
+ * ex.run_asynchronously(sequence, context);
+ *
+ * // Periodically call update() to bring our local copy of the sequence in sync with the
+ * // other thread. Once the sequence has finished, update() returns false and the thread
+ * // is joined automatically.
+ * while (ex.update(sequence))
+ *     sleep(0.1s);
  * \endcode
  *
- * \section usage Usage
+ * \section usage Usage Notes
  *
- * All functions and classes are declared in the namespace \ref task. To use the library,
- * include the single header file \link taskomat/taskomat.h \endlink and link your code
- * against both this library and the General Utility Library (-ltaskomat -lgul14).
+ * Taskolib requires at least C++17. All functions and classes are declared in the
+ * namespace \ref task. To use the library, include the single header file
+ * \link taskomat/taskomat.h \endlink and link your code against both this library and
+ * the General Utility Library (-ltaskomat -lgul14).
  *
  * The evolving design goals of the library are documented in a series of
  * \ref design_documents.
  *
- * \note
- * Taskolib requires at least C++17.
- *
- * \authors Lars Froehlich, Olaf Hensler, Ulf Fini Jastrow, Marcus Walla (for third-party
+ * \authors Lars Fröhlich, Olaf Hensler, Ulf Fini Jastrow, Marcus Walla (for third-party
```

Comment:
Are you sure you want this change? Generally C++ code files are not UTF8 (unlike Python3 for example); and this will break if the person viewing the file assumes a different encoding.

Reply:
I am reasonably sure that this is the right thing to do. This is not really a source file, but our template for the Doxygen documentation main page. On the HTML side, we have UTF-8 guaranteed. And if someone edits this with the wrong encoding, it should be obvious in code review. Fingers crossed... :)